packages/protocol-contracts/contracts/ERC20Custody.sol

```diff
@@ -0,0 +1,199 @@
+     * @param TSSAddress_, new tss address
+     */
+    function updateTSSAddress(address TSSAddress_) external onlyTSSUpdater {
+        if (TSSAddress_ == address(0)) {
+            revert ZeroAddress();
+        }
+        TSSAddress = TSSAddress_;
+    }
+
+    /**
+     * @dev Update zeta fee
+     * @param zetaFee_, new zeta fee
+     */
+    function updateZetaFee(uint256 zetaFee_) external {
+        if (msg.sender != TSSAddress) {
+            revert InvalidSender();
+        }
+        if (zetaFee_ == 0) {
+            revert ZeroFee();
+        }
+        if (zetaFee_ > zetaMaxFee) {
+            revert ZetaMaxFeeExceeded();
+        }
+        zetaFee = zetaFee_;
+    }
+
+    /**
+     * @dev Change the ownership of TSSAddressUpdater to the Zeta blockchain TSS nodes.
+     * Effectively, only Zeta blockchain validators collectively can update TSSAddress afterwards.
+     */
+    function renounceTSSAddressUpdater() external onlyTSSUpdater {
+        if (TSSAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        TSSAddressUpdater = TSSAddress;
+        emit RenouncedTSSUpdater(msg.sender);
+    }
+
+    /**
+     * @dev Pause custody operations.
+     */
+    function pause() external onlyTSSUpdater {
+        if (paused) {
+            revert IsPaused();
+        }
+        if (TSSAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        paused = true;
+        emit Paused(msg.sender);
+    }
+
+    /**
+     * @dev Unpause custody operations.
+     */
+    function unpause() external onlyTSSUpdater {
+        if (!paused) {
+            revert NotPaused();
+        }
+        paused = false;
+        emit Unpaused(msg.sender);
+    }
+
+    /**
+     * @dev Whitelist asset
+     * @param asset, ERC20 asset
+     */
+    function whitelist(IERC20 asset) external onlyTSS {
+        whitelisted[asset] = true;
+        emit Whitelisted(asset);
+    }
+
+    /**
+     * @dev Unwhitelist asset
+     * @param asset, ERC20 asset
+     */
+    function unwhitelist(IERC20 asset) external onlyTSS {
+        whitelisted[asset] = false;
+        emit Unwhitelisted(asset);
+    }
+
+    /**
+     * @dev Deposit asset amount to recipient with message that encodes additional zetachain evm call or message.
+     * @param recipient, recipient address
+     * @param asset, ERC20 asset
+     * @param amount, asset amount
+     * @param message, bytes message or encoded zetechain call.
+     */
+    function deposit(bytes calldata recipient, IERC20 asset, uint256 amount, bytes calldata message) external {
+        if (paused) {
+            revert IsPaused();
+        }
+        if (!whitelisted[asset]) {
+            revert NotWhitelisted();
+        }
+        if (zetaFee != 0 && address(zeta) != address(0)) {
+            zeta.transferFrom(msg.sender, TSSAddress, zetaFee);
+        }
+        uint256 oldBalance = asset.balanceOf(address(this));
```

Comment:
good catch to consider deflationary tokens!

Reply:
Audit report :)